### ISSUE TITLE
[Enhancement] Get multiple magnet links from one page 

### ISSUE THREAD

**Issue by icarok99** [i96751414/script.flix.magneto#17]
With this improvement it will be possible to acquire several magnet links from a single page and it will be possible to use providers that have several magnet links on a single page.

**Comment by i96751414** [comment]
This has been implemented and tested. I believe we can close this one.

**Comment by icarok99** [comment]
@i96751414 It didn't work as expected, there needs to be a subpage option and a method to scrape the subpage.

**Comment by icarok99** [comment]
> @i96751414 It didn't work as expected, there needs to be a subpage option and a method to scrape the subpage.

This way you will get several magnet links from the subpage

**Comment by i96751414** [comment]
What you mean? That's exactly what it is doing. Please have a look at the example I provided here: https://github.com/i96751414/script.flix.magneto/issues/11#issuecomment-1958036425

**Comment by icarok99** [comment]
> O que você quer dizer? Isso é exatamente o que está fazendo. Por favor, dê uma olhada no exemplo que forneci aqui: [# 11 (comentário)](https://github.com/i96751414/script.flix.magneto/issues/11#issuecomment-1958036425)

I used this example and it didn't work.

**Comment by icarok99** [comment]
![Screenshot_2024-02-29-16-44-28-447_org xbmc kodi](https://github.com/i96751414/script.flix.magneto/assets/46580318/69653807-4f12-41a7-b05c-7dc6ea5124a1)

![Screenshot_2024-02-29-16-44-55-954_pl solidexplorer2](https://github.com/i96751414/script.flix.magneto/assets/46580318/f85495b1-f774-4045-ab98-67f869b923dd)

**Comment by i96751414** [comment]
Are you sure you are using the latest version?

**Comment by icarok99** [comment]
I just checked with another film and it seems that not everyone has only 1 result.

![Screenshot_2024-02-29-17-02-34-796_org xbmc kodi](https://github.com/i96751414/script.flix.magneto/assets/46580318/7052ad56-deeb-4cbb-8e41-fe43fb03e8a0)

**Comment by i96751414** [comment]
"Wonka" for instance should show 5 results

**Comment by i96751414** [comment]
Ok, I found the problem. The scraper is working just fine, but that the Provider is filtering out 4 of 5 results.

**Comment by i96751414** [comment]
Fixed, now supporting base32 info hashes as well: 7557dbfdea59a6ce9b81fc0573214ba5ec0b84c4

I assume that the website you were scraping were using base32 hashes

**Comment by icarok99** [comment]
@i96751414 Is there any way I can use this style of parser?

```        "parser": {
            "infohash": "",
            "title": "item('h1', order=1)",
            "peers": "-1",
            "row": "find_all('div', ('class','box-filme-item'))",
            "seeds": "-1",
            "size": "",
            "magnet": "item(tag='a', attribute='href', order=1)"
        },

**Comment by icarok99** [comment]
I find it more simplified and easier to add providers

**Comment by i96751414** [comment]
No. This uses python's builtin xpath support. I'm not sure what are those expressions to be honest

**Comment by icarok99** [comment]
> No. This uses python's builtin xpath support. I'm not sure what are those expressions to be honest

AND ehp.py

**Comment by icarok99** [comment]
![IMG_20240302_200744](https://github.com/i96751414/script.flix.magneto/assets/46580318/1db993b1-d54e-424e-9d86-8fada11c01d0)

**Comment by icarok99** [comment]
@i96751414 
https://elementumorg.github.io/burst/create/
https://github.com/iogf/ehp
